### ISSUE TITLE
[sc-ide] zoom actions for all tabs

### ISSUE THREAD

**Issue by gusano** [supercollider/supercollider#3876]
As mentioned in #3863 it would be a great improvement to have zoom actions applied to all opened tabs instead of only the current one.

### Current behavior

Hit `CTRL` + `+` (`CMD` + `+` on mac) several times to increase the font size (zoom in), current tab has bigger font size while all other tabs don't

### Expected behavior

Hit `CTRL` + `+` (or `CMD` + `+` on mac) several times increase the font size for all tabs

**Comment by prko** [comment]
I think that the current behaviour can be preserved, and your suggestion can be added.
` CTRL (or CMD)` + `+`: zoom in current tab
` CTRL (or CMD)` + `-:` zoom out current tab
` CTRL (or CMD)` + `ALT` + `+`: zoom in All documents
` CTRL (or CMD)` + `ALT` + `-`: zoom out All documents

**Comment by gusano** [comment]
@prko IMHO it's an overkill to add yet another shortcut/action while my suggestion sounds like a simpler improvement (and this is the behavior on other text editors as well)

**Comment by brianlheim** [comment]
I agree it's simpler for everyone if this is a cross-document setting only.

**Comment by jamshark70** [comment]
So let's say I'm practicing live coding in one window, where I'm using a larger font because that's what I would use on stage, and I hit a bug in the support code, so I investigate in another document window, where I want to be able to read more code at one time (smaller font).

I don't know what it is about this particular issue, but just two days ago, it was "you shouldn't want to change all document windows' font sizes at once" and now it's "you shouldn't want to change them individually."

Maybe the IDE should help users get what they want instead of prescribing the one and only correct way people should view code.

**Comment by gusano** [comment]
> I don't know what it is about this particular issue, but just two days ago, it was "you shouldn't want to change all document windows' font sizes at once" and now it's "you shouldn't want to change them individually."

@jamshark70 Either I am missing something or you misread the original issue mentioned above? Where was mentioned that *"you shouldn't want to change all document windows' font sizes at once"*?
The issue @telephon brought up was that you need to increase/decrease the font size *for every document one by one manually* when using keyboard shortcut, which IMHO is a pain (`scide` is the only known editor which does that, all other do zoom in/out for all opened document).

> Maybe the IDE should help users get what they want instead of prescribing the one and only correct way people should view code

I guess that means you disagree with the improvement proposal?

**Comment by jamshark70** [comment]
I'll take a full reply to the mailing list, as it isn't strictly about the technical matter at hand.

I do think it should be both possible and easy to have separate font sizes per document tab, *and* it should be easy to change all of them at once. In that sense, I disagree with the proposal here if it discards the former.

If keyboard shortcuts zoom everything, and ctrl-scroll is local to the current tab, then IMO problem solved.

**Comment by jamshark70** [comment]
Actually, I'll drop it. My characterization of the other issue is exaggerated somewhat -- I apologize for that, while noting that it was fueled by one comment in particular that seemed to me to dismiss the usage scenario without sufficient consideration. But, it's not worth rehashing at length.

I can anticipate that it would be useful for me, in some cases, to have one tab for classroom-visible code, but maybe I need to check something quickly in a class definition, and it isn't important for the second window's text to be large (it's better if it isn't large).

It looks to me like the proposal is to enforce all tabs to have the same size. What I was trying to say is, we don't know if that's right for all cases and I'm not convinced by the discussion above that we've given sufficient consideration to other cases. It looks to me like "let's just totally reverse it," without asking anybody how it would affect them.

I'm not convinced by other editors' behavior. I'm also not convinced by the argument about interface complexity. If SC users can't handle a distinction between "enlarge all tabs" vs "enlarge the current tab," they're not going to get very far with the language.

A technical objection to the use case I proposed is: When you open a document, do you want the default font size or the last-set font size? I can imagine people wanting both in different circumstances. *That* is a kind of complexity that is harder to handle (and might be the nail in the coffin for my idea :stuck_out_tongue_winking_eye: )

**Comment by telephon** [comment]
If the semantics of cmd-+ / - is "zoom" and not font size, then this is very good. The font size is part of the scide preferences.

On the long run, the actual font size could become part of the document itself in a a normal rich text editor.

**Comment by htor** [comment]
applying zoom to all tabs sounds like a good idea. personally, this is what i'm used to from other IDE's, editors and my terminal emulator.

> I can anticipate that it would be useful for me, in some cases, to have one tab for classroom-visible code, but maybe I need to check something quickly in a class definition, and it isn't important for the second window's text to be large (it's better if it isn't large).

i would argue that `ctrl/cmd+0` solves the use case of switching context where you need to quickly reset the font size to normal.

**Comment by jamshark70** [comment]
> i would argue that ctrl/cmd+0 solves the use case of switching context where you need to quickly reset the font size to normal.

So, you've reset one tab to default size while the others are at the larger size. Only a matter of time before someone wants to re-enlarge that tab only, but it seems we're deciding not to provide a way to do that.

I'm beginning to accept that allowing different font sizes per tab introduces complications beyond just extra shortcuts. I remain unconvinced by the shortcuts argument, but I can imagine that other editors enforce one size for everything because the permutations of user needs get out of control, quickly.

So, ok, I withdraw my objection.

**Comment by gusano** [comment]
> So, you've reset one tab to default size while the others are at the larger size

The proposal implies that resets would also affect all tabs. Gonna edit the description.
